### PR TITLE
Replace a couple of diagrams with mermaid code.

### PR DESCRIPTION
**Pull request by cqc-alec**
WIP on #106 .

I got fed up trying to do the CFG, but will probably return to it (and the others).

I'm just getting to grips with mermaid and not entirely happy with the appearance of the conditional, but couldn't see how to force the case boxes to be side by side.

### CODE REVIEW

File: specification/hugr.md [hunk 1]
```diff
@@ -335,9 +364,22 @@ A **Predicate(T0, T1…TN)** type is an algebraic “sum of products” type,
 defined as `Sum(Tuple(#t0), Tuple(#t1), ...Tuple(#tn))` (see [type
 system](#type-system)), where `#ti` is the *i*th Row defining it.
 
-**TODO: update below diagram now that Conditional is “match”**
-
-<img src="attachments/2647818241/2647818344.png" width="1024px">
+```mermaid
+flowchart
```
Comment (ss2165) [inline specification/hugr.md:368]: is there  a way to get the cases to appear side by side?

Comment (cqc-alec) [reply specification/hugr.md:368]: I wanted to do that but couldn't find a way.

Comment (ss2165) [reply specification/hugr.md:368]: oh sorry for not reading the PR description properly

File: specification/hugr.md [hunk 2]
```diff
@@ -335,9 +364,22 @@ A **Predicate(T0, T1…TN)** type is an algebraic “sum of products” type,
...
+    subgraph Conditional
+        subgraph Case0["Case 0"]
+            C0I["case 0 inputs + other inputs"] --> op0["operations"]
+            op0 --> C0O["outputs"]
+        end
+        subgraph Case1["Case 1"]
+            C1I["case 1 inputs + other inputs"] --> op1["operations"]
+            op1 --> C1O["outputs"]
+        end
+    end
```
[the remark this answers is not in the document]
Comment (cqc-alec) [reply specification/hugr.md:378]: > Can be done with subgraph direction + invisible link
> 
> https://mermaid.js.org/syntax/flowchart.html#an-invisible-link

Oh, nice, thanks!

**Review by ss2165 — APPROVED** [review]

**Pull request merged**